zk/stages/stage_sequence_execute_transactions.go

```diff
@@ -149,6 +137,14 @@ func attemptAddTransaction(
 
 	txCounters := vm.NewTransactionCounter(transaction, sdb.smt.GetDepth(), uint16(forkId), cfg.zk.VirtualCountersSmtReduction, cfg.zk.ShouldCountersBeUnlimited(l1Recovery))
 	overflow, err = batchCounters.AddNewTransactionCounters(txCounters)
+	if err != nil {
+		if overflow && !l1Recovery {
```

Comment:
we need to check the `overflow` bool outside of the `err` check.  An overflow might not trigger an error state and we'll miss it when we need to handle it.

Reply:
I reverted this part